packages/transaction-controller/src/TransactionController.ts

```diff
@@ -1338,24 +1359,25 @@ export class TransactionController extends BaseControllerV1<
       ),
     });
 
-    const etherscanRemoteTransactionSource =
-      new EtherscanRemoteTransactionSource({
-        includeTokenTransfers:
-          this.incomingTransactionOptions.includeTokenTransfers,
-      });
     const incomingTransactionHelper = new IncomingTransactionHelper({
       blockTracker: networkClient.blockTracker,
       getCurrentAccount: this.getSelectedAddress,
       getLastFetchedBlockNumbers: () => this.state.lastFetchedBlockNumbers,
       // TODO(JL): Fix this type
+      // TODO (AD):
+      // This is a hack until we remove the base IncomingTransactionHelper class
+      // and should be replaced with a plain chainId parameter
       getNetworkState: () => {
         return {
           providerConfig: { chainId } as ProviderConfig,
         } as NetworkState;
       },
       isEnabled: this.incomingTransactionOptions.isEnabled,
       queryEntireHistory: this.incomingTransactionOptions.queryEntireHistory,
-      remoteTransactionSource: etherscanRemoteTransactionSource,
+      remoteTransactionSource:
+        // we know that the map has a value for this chainId because we just set it above
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        this.etherscanRemoteTransactionSourcesMap.get(chainId)!,
```

Comment:
not sure if this is worth doing to avoid having to disable this eslint rule

```
  let etherscanRemoteTransactionSource = this.etherscanRemoteTransactionSourcesMap.get(chainId)
  if (!etherscanRemoteTransactionSource) {
    etherscanRemoteTransactionSource =
      new EtherscanRemoteTransactionSource({
        includeTokenTransfers:
          this.incomingTransactionOptions.includeTokenTransfers,
      });
    this.etherscanRemoteTransactionSourcesMap.set(
      chainId,
      etherscanRemoteTransactionSource,
    );
  }


  ...
  remoteTransactionSource: etherscanRemoteTransactionSource,
  ...
```

Reply:
I can work with this

Reply:
done here: https://github.com/MetaMask/core/pull/3822/commits/00f5192cff537500fa217bd755e90f733746d878